components/ILIAS/Mail/classes/Message/MailBoxQuery.php

```diff
@@ -0,0 +1,386 @@
+        $this->db = $DIC->database();
+    }
+
+    public function withFolderId(?int $folder_id): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->folder_id = $folder_id;
+        return $clone;
+    }
+
+    public function withSender(?string $sender): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->sender = $sender;
+        return $clone;
+    }
+
+    public function withRecipients(?string $recipients): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->recipients = $recipients;
+        return $clone;
+    }
+
+    public function withSubject(?string $subject): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->subject = $subject;
+        return $clone;
+    }
+
+    public function withBody(?string $body): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->body = $body;
+        return $clone;
+    }
+
+    public function withIsUnread(?bool $is_unread): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->is_unread = $is_unread;
+        return $clone;
+    }
+
+    public function withIsSystem(?bool $is_system): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->is_system = $is_system;
+        return $clone;
+    }
+
+    public function withHasAttachment(?bool $has_attachment): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->has_attachment = $has_attachment;
+        return $clone;
+    }
+
+    public function withPeriodStart(?DateTimeImmutable $period_start): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->period_start = $period_start;
+        return $clone;
+    }
+
+    public function withPeriodEnd(?DateTimeImmutable $period_end): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->period_end = $period_end;
+        return $clone;
+    }
+
+    /**
+     * @param int[]|null $filtered_ids
+     */
+    public function withFilteredIds(?array $filtered_ids): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->filtered_ids = $filtered_ids;
+        return $clone;
+    }
+
+    public function withLimit(int $limit): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->limit = $limit;
+        return $clone;
+    }
+
+    public function withOffset(int $offset): MailBoxQuery
+    {
+        $clone = clone $this;
+        $clone->offset = $offset;
+        return $clone;
+    }
+
+    public function withOrderColumn(?MailBoxOrderColumn $order_column): MailBoxQuery
+    {
+        $clone = clone $this;
+        if (isset($order_column)) {
+            $clone->order_column = $order_column;
+        }
+        return $clone;
+    }
+
+    public function withOrderDirection(?string $order_direction): MailBoxQuery
+    {
+        $clone = clone $this;
+        if (in_array($order_direction, [Order::ASC, Order::DESC])) {
+            $clone->order_direction = $order_direction;
+        }
+        return $clone;
+    }
```

Comment:
Same here
```suggestion
    public function withOrderDirection(string $order_direction): MailBoxQuery
    {
        if ($order_direction !== Order::ASC && $order_direction !==  Order::DESC) {
            throw new InvalidArgumentException('Invalid type of order');
        }
        $clone = clone $this;
        $clone->order_direction = $order_direction;
        return $clone;
    }
```

Reply:
Solved by applying the default order direction when null is given.